p2p/transport.go

```diff
@@ -261,6 +261,12 @@ func (mt *MultiplexTransport) Listen(addr NetAddress) error {
 	return nil
 }
 
+// AddChannel registers a channel to nodeInfo. This will be broadcaster to peers
+// to make them aware of what channels are open
+func (mt *MultiplexTransport) AddChannel(chID byte) error {
+	return mt.nodeInfo.(DefaultNodeInfo).AddChannel(chID)
```

Comment:
is this safe?

Reply:
what sort of scenario do you have in mind?

Reply:
The problem I see is that I've designed this to only be done on startup whereas I'm unsure on the affect if this is done whilst the node is running

Reply:
oh, just type casting, without a check, always worries me a bit.